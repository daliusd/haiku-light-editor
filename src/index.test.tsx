import React from 'react'

import { Editor } from '.'
import { render, screen } from '@testing-library/react'

describe('Editor', () => {
  it('is truthy', () => {
    expect(Editor).toBeTruthy()
  })

  const settings = {
    width: 100,
    height: 100
  }

  const editable = {
    width: 20,
    height: 20
  }

  it('renders empty square', () => {
    render(<Editor settings={settings} editable={editable} />)
    expect(screen.getByTestId('editor')).toMatchSnapshot()
  })

  it('renders empty square in black background', () => {
    render(
      <Editor
        settings={{ ...settings, backgroundColor: '#000000' }}
        editable={editable}
      />
    )
    expect(screen.getByTestId('editor')).toMatchSnapshot()
  })
})
