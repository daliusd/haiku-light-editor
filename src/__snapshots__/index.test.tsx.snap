// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor renders empty square 1`] = `
<div
  class="editor"
  data-testid="editor"
  style="width: 100px; height: 100px; background: rgb(238, 238, 238);"
>
  <div
    style="width: 80px; height: 80px; left: 10px; top: 10px; position: absolute; background: rgb(255, 255, 255);"
  />
</div>
`;

exports[`Editor renders empty square in black background 1`] = `
<div
  class="editor"
  data-testid="editor"
  style="width: 100px; height: 100px; background: rgb(0, 0, 0);"
>
  <div
    style="width: 80px; height: 80px; left: 10px; top: 10px; position: absolute; background: rgb(255, 255, 255);"
  />
</div>
`;
